pkg/api/pluginproxy/ds_proxy.go

```diff
@@ -42,6 +42,9 @@ type httpClient interface {
 
 func NewDataSourceProxy(ds *m.DataSource, plugin *plugins.DataSourcePlugin, ctx *m.ReqContext, proxyPath string) *DataSourceProxy {
 	targetURL, _ := url.Parse(ds.Url)
+	if targetURL.Scheme == "unix" {
+		targetURL, _ = url.Parse("http://localhost")
```

Comment:
why does this force localhost?   (I don't really know how this works, but it looks weird)

Reply:
Guess you can only connect to unix sockets on the local machine, that's one of the good thing with using them from a security perspective. Still though require you to run all your TSDB's and Grafana on the same machine which seems quite limited.

Reply:
@marefr That's not exactly true, for example in case of Docker, sockets can be shared with the host or other containers.

Reply:
again -- i don't really know how this works, but if it should only work on localhost we should throw an error if the host is not localhost rather than just switching to it.

Reply:
@ryantxu unix schemes (like unix:///var/run/mysqld/mysqld.sock) does not have any host.
HTTP requires a "host" even when sending on unix socket. Thats why I am forcing it to use localhost. Doesnt really matter what we set it to. It does not route to that address.

@Majkl578 i agree sockets can be shared but only people with access to host can have access sockets. More secured than exposing ports (as @marefr mentioned)

Reply:
can we add a comment that explains that?

Since this modifies the proxy for all datasource, where is the best place to document how this feature can be used?

Reply:
best place will be datasource documentation. (Thats where i would look)
Yes, it proxies for all data source. Best example is mysql.

There would also be additions to whitelist documentation.
unix schemes do not have host. we can restrict them to certain paths (eg /var/run/mysqld/mysqld.sock )